### DISCUSSION TITLE
Azure Private Link Service + Data Factory

### DISCUSSION THREAD

**Discussion by maciejgos** [AzureCommunityPL/group#324] [Pomoc]
Cześć czy ktoś z was konfigurował połączenie jak w tytule? 

Próbuje zrobić konfiguracje jak w artykule https://docs.microsoft.com/en-us/azure/data-factory/tutorial-managed-virtual-network-on-premise-sql-server niestety mam totalny brak ruchu na Private Link Service. 

Ruch przechodzi przez Load Balancer i DNS Forwarder, oraz mam komunikacje po porcie 1433. Udało mi się nawet dostać z VM do instancji SQL Server w on-prem poprzez wszystkie wymienione elementy. Problemem jest tylko połączenie ADF i Link Service.

Spotkaliście się z takim problemem? Może macie jakieś doświadczenia w diagnozowaniu tego typu problemu?

**Comment by TomaszOledzki** [comment]
Dla pewności zapytam. Jaki masz Integration Runtime?

**Reply by maciejgos** [reply]
AutoResolve IR z Managed VNET

**Comment by krukowskid** [comment]
Całkiem niedawno to konfigurowałem
ADF z private endpoint w tym samym VNET, w którym jest karta sieciowa z IR (vm w azure). W ADF integration runtime ustawiony na VM i wszystko śmiga bez problemu. 
Z poziomu IR test do sql przechodzi?

**Reply by maciejgos** [reply]
Ale to masz na myśli Self-Hosted IR? Nie do końca łapię "ustawienie IR na VM w Azure"?

Ja u siebie korzystam z AutoResolve IR z ustawionym Managed VNET i niestety ale testy nie przechodzi do SQL Server.

**Reply by krukowskid** [reply]
OK. Myślałem, że pisząc o VM masz na myśli self-hosted.  Wydaje mi się, że z IR z Azure nie ma możliwości dostać się do zasobów, które nie są dostępne z poziomu internetu. 
edit: muszę jeszcze raz wszystko przeczytać, bo źle zrozumiałem :)
edit2: to tak niestety nie robiłem. Zestawiałem tylko tak jak pisałem wcześniej ADF w ramach sieci prywatnej z private endpoints i self-hosted IR w azure. 
Próbowałeś dodać private DNS zone, zrobić tam wpis na SQL i podpiąć pod vnet w którym jest ten subnet?

**Comment by krukowskid** [comment]
@maciejgos jestem ciekawy jak udało się rozwiązać ten problem

**Reply by maciejgos** [reply]
Tak udało się uruchomić to połączenie. Wymagane do tego było skonfigurowanie Private DNS Zone i upewnienie się, że DNS się rozpropagował. Dodatkowo po ostatnich zmianach w ADF dla Private Link Service trzeba użyć tylko FQDN bez NAT IP.

Po wszystkich zabiegach i kilku call z Indiami zadziałało.
